src/DynamoCoreWpf/ViewModels/Search/SearchViewModel.cs

```diff
@@ -328,19 +330,81 @@ private void InsertEntry(NodeSearchElementViewModel entry, IEnumerable<string> c
             var nameStack = new Stack<string>(categoryNames.Reverse());
             var target = libraryRoot;
             bool isRoot = true;
+            string path = "";
+            ClassesNodeCategoryViewModel targetClass = null;
             while (nameStack.Any())
             {
                 var next = nameStack.Pop();
+                path = AddToFullPath(path, next);
+
                 var categories = target.SubCategories;
-                var newTarget = categories.FirstOrDefault(c => c.Name == next);
+                NodeCategoryViewModel targetClassSuccessor = null;
+                var newTarget = categories.FirstOrDefault(c =>
+                {
+                    // Each path has one class. We should find and save it.                    
+                    if (c is ClassesNodeCategoryViewModel)
+                    {
+                        targetClass = c as ClassesNodeCategoryViewModel;
+                        // As soon as ClassesNodeCategoryViewModel is found we should search 
+                        // through all it classes and save result.
+                        targetClassSuccessor = c.SubCategories.FirstOrDefault(c2 => c2.Name == next);
+                        return targetClassSuccessor != null;
+                    }
+
+                    return c.Name == next;
+                });
                 if (newTarget == default(NodeCategoryViewModel))
                 {
                     newTarget = isRoot ? new RootNodeCategoryViewModel(next) : new NodeCategoryViewModel(next);
```

Comment:
For the above condition, do turn it into this:

```csharp
newTarget = ((target == libraryRoot) ?
    new RootNodeCategoryViewModel(next) :
    new NodeCategoryViewModel(next);
```

This way we can remove `isRoot` and reduce the chances for problem (e.g. if for some code path we forgot to reset the value of `isRoot`).